dvc/output/base.py

```diff
@@ -195,17 +196,35 @@ def get_hash(self):
                 self.path_info,
                 self.fs,
                 self.fs.PARAM_CHECKSUM,
+                dvcignore=self.dvcignore,
             ).hash_info
         return ostage(
-            self.odb, self.path_info, self.fs, self.odb.fs.PARAM_CHECKSUM
+            self.odb,
+            self.path_info,
+            self.fs,
+            self.odb.fs.PARAM_CHECKSUM,
+            dvcignore=self.dvcignore,
         ).hash_info
 
     @property
     def is_dir_checksum(self):
         return self.hash_info.isdir
 
+    def check_path_dvcignore(self, path) -> bool:
```

Comment:
I think we can make it with underscore (`_check_path_dvcignore`), so that its not exposed outside, since we are using it internally for now.